internal/functions/Get-XpDirTreeRestoreFile.ps1

```diff
@@ -42,7 +42,7 @@ function Get-XpDirTreeRestoreFile {
     Write-Message -Level Verbose -Message "Connecting to $SqlInstance"
     $server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
 
-    if (($path -like '*.bak') -or ($path -like '*trn')) {
+    if ((($path -like '*.bak') -or ($path -like '*.trn')) -and (Test-Path -Path $path -PathType Leaf)) {
```

Comment:
`Test-Path` is only valid for local path testing. Test-DbaSqlPath would be valid, but the path is likely already tested in another portion of the script (please check). Also, can you confirm that -like isn't interpreting '.' as a special case?

Thanks again for the PR 👍

Reply:
You're right. Test-Path is incorrect in this case. Sorry I did not tested it.
The idea was to verify if $path points to file, not directory.
I wanted to exclude the case when someone name database like "myDatabase.bak" which is totally confused but unfortunately allowed in SQL.

I'll try to find another solution for this issue.

Reply:
try `split-path` - should work 💃 

![image](https://user-images.githubusercontent.com/8278033/44547398-49dd5e00-a71b-11e8-86a4-db488f00bf2f.png)

Reply:
Unfortunately Split-Path does not work in this case (or I don't see it :-) )

I've found a solution to use Test-Path for shares:
![obraz](https://user-images.githubusercontent.com/42638464/44569237-fe0dd180-a779-11e8-867c-feedecfc71ef.png)


If it's acceptable solution, please let me know how should I proceed with this change. New pull request? Or change this one? I'm newbie in this topic :-)

Reply:
Thanks for putting so much thought into this 🤗 Awesomely enough, GitHub allows you to just update the branch with the PR and it'll update the PR.

Unfortunately, Test-Path will not do the trick. What if your SQL Server account has access but you do not? I'm not sure there is a good solution to this, but I think that .trn is still useful.

Reply:
Oh, now I got why there's so many `xp_dirtree` and `xp_fileexists` in dbatools code :-)

Challenge accepted :-)

BTW: awesome work with the whole dbatools solution. It helps me a lot.

Reply:
thank you :D glad you find it useful.W ould you like to add to . and then submit another PR once you figure out the testing?